Made the support files (gemspec, Gemfile, etc) look like goliath's)

mrflip opened the pull request:
Going to add some stuff, so first putting the house in order.

Commit:
Made the support files (gemspec, Gemfile, etc) look like goliath's)

igrigorik commented on goliath-contrib.gemspec:
```diff
@@ -1,23 +1,61 @@
+  s.add_dependency 'goliath'
 
-  gem.add_dependency 'goliath'  
+  s.add_development_dependency 'rspec', '>2.0'
```
I don't think we need to replicate all the gems here? As long as we require goliath gem itself, we should inherit all the right deps.. no? 

Otherwise we'll just fall out of sync.

mrflip replied:
The ones I replicated were the `development_dependency`s, for Bundler's benefit. I don't have a good understanding the interplay among an app's `Gemfile`, its gemspec's `development_dependency` and `dependency`s, and the  `development_dependency`/`dependency`s of the gems it in turn specifies, but my cargo-cult ritual is to put them in either the Gemfile or the gemspec as development dependencies.

I can confirm though that commenting eg `tilt` out of the gemspec (but leaving the `goliath` dependency) means that bundler doesn't include it:

```
# after commenting out that line in the gemspec
worblehat ~/ics/ext/goliath-contrib$ bundle exec ruby -e 'require "tilt"'
-e:1:in `require': cannot load such file -- tilt (LoadError)
    from -e:1:in `<main>'
```

Anyway: I think things like `em-http-request` aren't brought over by the simple dependency on goliath, so they need to either be here or in the gemfile.  I put it here because I figured it could thus be easily diff'ed (which is also why I brought over the full list, not just the ones the examples referenced).

igrigorik replied:
If we're moving middleware pieces out of core into contrib than then deps (including test libs) should be referenced here (in one place max). 

I don't think we need nokogiri, etc, here.. unless we're explicitly using these libs in the contrib middleware.

Commit:
don't require goliath/contrib in goliath/contrib

Commit:
Rakefile does yard & tests too. Dependency list more minimalist.
The rakefile is copied from goliath; it now has tasks for running tests, making yardoc, etc

mrflip commented:
Last commit knocks the list back to only what's needed now by code or for spec/yard/gem generation. The maruku/bluecloth are needed by YARD. EM::HTTP and postrank-uri are used by son_of_a_batch.
